### PR TITLE
add to the cart functionality

### PR DESCRIPTION
**Pull request by rishabhharod**

### CODE REVIEW

File: app/controllers/carts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CartsController < ApplicationController
+  before_action :set_cart,only: [:show]
+  def show
+    @cart_items = @cart.cart_items
+  end
+  private
```
Comment (vinayaklovanshi) [inline app/controllers/carts_controller.rb:6]: Add space above and below the private methods

File: .byebug_history [hunk 2]
```diff
@@ -1,4 +1,75 @@
 c
+@cart.cart_items.find(params[:id])
```
Comment (vinayaklovanshi) [inline .byebug_history:2]: No need to push byebug history

File: app/controllers/cart_items_controller.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+class CartItemsController < ApplicationController
+  before_action :set_cart
+  def create
+    product = Product.find(params[:product_id])
```
Comment (vinayaklovanshi) [inline app/controllers/cart_items_controller.rb:4]: Add it in before_action

File: app/controllers/cart_items_controller.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    @cart_item = @cart.cart_items.find_or_initialize_by(product_id: product.id)
+    if @cart_item.new_record?
+      @cart_item.quantity = 1
+    else
+      @cart_item.quantity += 1
+    end
+    if @cart_item.save
+      flash[:notice] = "Product added to the cart."
```
Comment (vinayaklovanshi) [inline app/controllers/cart_items_controller.rb:12]: Avoid " " user single ' '

File: app/controllers/carts_controller.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class CartsController < ApplicationController
+  before_action :set_cart,only: [:show]
+  def show
```
Comment (vinayaklovanshi) [inline app/controllers/carts_controller.rb:3]: Add space above

**Review by vinayaklovanshi — CHANGES_REQUESTED** [review]

File: app/controllers/cart_items_controller.rb (hunk 4 above)
Comment (rishabhharod) [reply app/controllers/cart_items_controller.rb:12]: okay sir

File: app/controllers/cart_items_controller.rb (hunk 3 above)
Comment (rishabhharod) [reply app/controllers/cart_items_controller.rb:4]: done sir

File: .byebug_history (hunk 2 above)
Comment (rishabhharod) [reply .byebug_history:2]: okay sir

**Pull request merged**